src/controllers/controller.h

```diff
@@ -136,6 +139,11 @@ class Controller : public QObject {
     const RuntimeLoggingCategory m_logInput;
     const RuntimeLoggingCategory m_logOutput;
 
+  public slots:
+    // This must be reimplemented by sub-classes desiring to send raw bytes to a
+    // controller.
+    virtual void sendBytes(const QByteArray& data) = 0;
```

Comment:
So this should just be defined in `public`? My understanding is this is needed so you can reference `&Controller::sendBytes` when doing a connect later, with child classes

Reply:
no its not required from what I know. It's a relic of old Qt versions being limited to very old compilers. `&Controller::sendBytes` is simply just a "pointer-to-member-function" which you can use with connect since a while, I think the `slots` directive was only needed for the macro+string based dispatch (see `SLOT()` macro).

Reply:
The rules might be a little different for virtual slots, but I don't think so.